src/data/program_configuration.js

```diff
@@ -312,20 +312,13 @@ class CrossFadedCompositeBinder implements AttributeBinder {
     zoomOutPaintVertexBuffer: ?VertexBuffer;
     paintVertexAttributes: Array<StructArrayMember>;
 
-    constructor(expression: CompositeExpression, names: Array<string>, type: string, useIntegerZoom: boolean, zoom: number, PaintVertexArray: Class<StructArray>, layerId: string) {
+    constructor(expression: CompositeExpression, type: string, useIntegerZoom: boolean, zoom: number, PaintVertexArray: Class<StructArray>, layerId: string) {
         this.expression = expression;
         this.type = type;
         this.useIntegerZoom = useIntegerZoom;
         this.zoom = zoom;
         this.layerId = layerId;
 
-        this.paintVertexAttributes = names.map((name) => ({
-            name: `a_${name}`,
-            type: 'Uint16',
-            components: 4,
-            offset: 0
```

Comment:
The issue lies here, instead of using `0`, we need to use `patternAttributes.members` which has the correct offsets and sizes.